libraries/AP_Logger/LoggerMessageWriter.cpp

```diff
@@ -233,14 +233,28 @@ void LoggerMessageWriter_WriteSysInfo::process() {
         stage = Stage::RC_PROTOCOL;
         FALLTHROUGH;
 
-    case Stage::RC_PROTOCOL:
+    case Stage::RC_PROTOCOL: {
         const char *prot = hal.rcin->protocol();
         if (prot == nullptr) {
             prot = "None";
         }
         if (! _logger_backend->Write_MessageF("RC Protocol: %s", prot)) {
             return; // call me again
         }
+        stage = Stage::BATT_CYCLES;
+        FALLTHROUGH;
+    }
```

Comment:
No need to close brackets here

Reply:
Yeah I'm not sure why my complier requires those scoping brackets? Here is what happens on my compiler without them.
![compiler_error](https://user-images.githubusercontent.com/69225461/103244009-99567900-4929-11eb-9d7a-f540acf93d5d.PNG)

Reply:
You're introducing a variable within a `case` - that requires the braces.

Reply:
OK, so on re-reading you're not actually doing that, but by falling through from the RC_PROTOCOL stage you're introducing a jump over the definition of its local variables.  The problem there is that if the switch dumps you straight into `BATT_CYCLES`  then the initialisation of the locals in the `RC_PROTOCOL` stage isnt done - but they're still in scope for `BATT_CYCLES`!

Yay compilers saving us from ourselves.

Reply:
Ahh nice thanks for that! I changed it a bit in the other pull request so those local variables are initialized before the switch and scoping isn't required.